### PR TITLE
Add device specific piggybacked responses.

### PR DESCRIPTION
**Pull request by boaks**
Legacy support for tenants with devices where old versions fails to
process piggybacked responses. Enable piggybacked responses, if device
indicates a newer version succeeding to process piggybacked responses
using the URI-query-parameter "piggy".

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
One alternative may be:
Use the "piggy" parameter only together with a tenant specific "timeout-to-ack", if that value is less then -1, e.g. "-750". That would enable a tenant specific timeout in combination with the "piggy" parameter. The change is a simple one.

Does the tenant-specific configuration allow negative values?

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java [hunk 1]
```diff
@@ -320,6 +331,18 @@ public Integer getCommandResponseStatus() {
...
 
+    /**
+     * Check, if request is sent from a device, which version supports piggybacked response.
+     *
+     * Overwrite a tenant specific disabled {@link CoapConstants#TIMEOUT_TO_ACK}.
+     *
+     * @return {@code true}, if device version supports piggybacked responses, {@code false}, for devices with "legacy
+     *         version", which doesn't support that.
+     */
+    public boolean isSupportingPiggyBacked() {
```
Comment (sophokles73) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java:342]: given that this is supposed to be an undocumented feature we should probably keep this method private

**Comment by boaks** [comment]
> Right?

Yes.

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java (hunk 1 above)
Comment (boaks) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java:342]: OK

Comment (boaks) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java]: Done.

File: adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java [hunk 2]
```diff
@@ -45,6 +45,11 @@ public final class CoapContext extends MapBasedTelemetryExecutionContext {
      * The query parameter which is used to indicate an empty notification.
      */
     public static final String PARAM_EMPTY_CONTENT = "empty";
+    /**
+     * The query parameter which is used to indicate, that a piggypacked response is supported by the device.
+     * (Legacy support for device with firmware versions not supporting  piggypacked response.)
+     */
+    public static final String PARAM_PIGGYBACKED = "piggy";
```
Comment (boaks) [inline adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java:52]: I can adapt this to `private` as well. Currently it's used in the unit tests, but I could copy it there, if it should be converted into `private`.

Comment (sophokles73) [reply adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java:52]: then maybe use package private scope?

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**